src/firefly/js/visualize/XYPlot.jsx

```diff
@@ -216,6 +235,8 @@ export class XYPlot extends React.Component {
                     chart.get(HIGHLIGHTED).setData(highlightedData);
                 }
 
+
+                // plot parameters change
```

Comment:
adding simple comments like these in this function really helps.  it was hard to read without it.